compiler/vm.nim

```diff
@@ -1035,7 +1035,13 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
       regs[ra].intVal = not regs[rb].intVal
     of opcEqStr:
       decodeBC(rkInt)
-      regs[ra].intVal = ord(regs[rb].node.strVal == regs[rc].node.strVal)
+      var ret = false
```

Comment:
cstring, string share many magic functions: mAppendStrStr, mLengthStr, mSlice.
All these functions now need to handle null as a value.
It would be more efficient to generate different code in vmgen to avoid if statements in vm rawExecute which is important to be fast.

Reply:
Likely it means that cstring and string need to get different op codes `opcEqStr`, `opcEqCStr`, `opcLenStr` and `opcLenCStr` and etc